bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java

```
@@ -612,6 +612,33 @@ public static String buildEobId(ClaimType claimType, String claimId) {
 		return String.format("%s-%s", claimType.name().toLowerCase(), claimId);
 	}
 
+	/**
+	 * @param eob
+	 *            the {@link ExplanationOfBenefit} to extract the id from
+	 * @return the <code>claimId</code> field value (e.g. from
+	 *         {@link CarrierClaim#getClaimId()})
+	 */
+	static String getUnprefixedClaimId(ExplanationOfBenefit eob) {
+		String id = null;
+		for (Identifier i : eob.getIdentifier()) {
+			if (i.getSystem().contains("clm_id") || i.getSystem().contains("pde_id")) {
+				id = i.getValue();
```

For robustness' sake, I'd make this an early return. Then, instead of returning at the end of the loop, throw a `BadCodeMonkeyException` since an ID was expected but none was found.

---

Done.